integration-core/src/main/java/org/symphonyoss/integration/core/bridge/StreamServiceImpl.java

```diff
@@ -37,44 +43,58 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 import javax.annotation.PostConstruct;
 
 /**
+ * Service component responsible to post message through the Agent Message API.
  * Created by rsanchez on 13/05/16.
  */
 @Component
 public class StreamServiceImpl implements StreamService {
 
   private static final Logger LOG = LoggerFactory.getLogger(StreamServiceImpl.class);
 
+  private static final String AGENT_SERVICE_NAME = "Agent";
+
+  private static final String AGENT_API_V3 = "1.46.0";
+
   @Autowired
   private AuthenticationProxy authenticationProxy;
 
   @Autowired
-  private AgentApiClient agentApiClient;
+  private AgentApiClient agentV2ApiClient;
```

Comment:
It works because AgentApiClient is a prototype component. I'll have two copies from the AgentApiClient. Each copy will be configured to be used with a specific version of the Message API.